freertos-rust/src/shim.rs

```diff
@@ -38,6 +38,15 @@ extern "C" {
     pub fn freertos_rs_give_mutex(mutex: FreeRtosQueueHandle) -> FreeRtosBaseType;
     pub fn freertos_rs_give_recursive_mutex(mutex: FreeRtosQueueHandle) -> FreeRtosBaseType;
 
+    pub fn freertos_rs_take_semaphore_isr(
+        semaphore: FreeRtosQueueHandle,
```

Comment:
I see the other semaphore functions take and emit QueueHandle but that looks wrong to me. The underlying type is the same, so it works, but I think it should be `FreeRtosSemaphoreHandle`. Changing this might be more than you wanted to do here so I'm happy to move that rework to a separate PR.

Reply:
Added issue as #54 54